sharedbuffers/mapped_struct.py

```diff
@@ -1001,46 +1007,51 @@ def unpack_from(cls, buf, offs, idmap = None, array = array.array, klass = list)
                 offs += 8
             w = buf[offs:offs + itemsize * objlen]
             if python3:
-                w = w.cast(dtype)
-            rv = array(dtype, w)
+                rv = w.cast(dtype)
+            else:
+                rv = array(dtype)
+                rv.frombytes(w)
```

Comment:
Shouldn't it be

```python3
rv = array(dtype)
rv.frombytes(w)
```

for both py2 and py3?

Reply:
No, memoryviews can be iterated as an array, and have a .tolist method, and that's all we care about in this context. And rv.tolist is faster in py3 with memoryviews, so it's preferred as it avoids having to create the intermediate array.